Source/WTF/wtf/BumpPointerAllocator.h

```diff
@@ -123,9 +124,12 @@ class BumpPointerPool {
                 return nullptr;
         }
 
+        if (poolSize > remainingCapacity)
+            return nullptr;
```

Comment:
If the parameter is actually "remaining capacity" this check isn't needed. If it's actually a maximum why do we need to compute `poolSize` above? I think you could basically just do `PageAlloaction::allocate(roundUpToMultipleOf(maximumCapacity, std::max(static_cast<size_t>(MINIMUM_BUMP_POOL_SIZE), WTF::pageSize()))`. I also don't know what `minimumCapacity` is doing at that point either, FWIW.

Reply:
`minimumCapacity` is used for cases where the client wants to specify a minimum size to allocate for the pool.  Otherwise, the allocated pool (for growing the eval stack) might not fit the allocation request.

`remainingCapacity` is an accurate description for this pool allocation.  If we rename it to `maximumCapacity`, then the way to think of it is as the max capacity for remaining allocation.  Hence, `remainingCapacity` is the better name.

> If the parameter is actually "remaining capacity" this check isn't needed.
Why is this check not needed?  The point is that `PageAllocation::allocate` may succeed, but we don't want to allow it because we're bounding the memory use of the eval stack (as implemented by the chain of `BumpPointerPool`s.  Can you explain what you mean here?

Reply:
I agree once it's passed to the `BumpPointerPool` it's the remaining capacity but from the caller of this function's perspective it's the total capacity since the "current capacity" is 0. If I saw the signature `BumpPointerPool* BumpPointerPool::create(size_t remainingCapacity, size_t minimumCapacity = 0)` first I would be confused about what it means for something to "remain" for a thing that doesn't exist yet. Then in a void I would expect that the returned pool would happily allocate until I hit `remainingCapacity + minimumCapacity` but I actually get a pool that will allocate until `remainingCapacity` bytes are used. I don't think any such confusion exists if the function is `BumpPointerPool* BumpPointerPool::create(size_t maximumCapacity, size_t minimumCapacity = 0)`

Reply:
Oh I misunderstood this class. I thought this create was a public API not an internal one. Let me look again.